torch/distributed/_shard/sharded_tensor/api.py

```diff
@@ -685,6 +741,10 @@ def local_tensor(self) -> torch.Tensor:
 
     @classmethod
     def __torch_function__(cls, func, types, args=(), kwargs=None):
+        # Access attributes as is usually done on Tensors
+        if func.__name__ in ["__get__", "size"]:
+            return super().__torch_function__(func, types, args, kwargs)
```

Comment:
So I tried it by making it a separate op, the biggest issue here is that if we make it a separate op using `sharded_op_impl`, we couldn't call `super()` because the cls instance is not passed to the separate op function, we can still call `super()` inside a classmethod and it will point to the correct instance. So I guess we can keep these two here until we find a better way.

Reply:
Do we need super() here? Can't we just do this in the `sharded_op_impl`:

```
with torch._C.DisableTorchFunction():
  func(*args, **kwargs)
```

Reply:
yeah that's sth we can do, I just tried this approach and the difference is that we need to register all the property getters separately (i.e. `Tensor.shape.__get__`, `Tensor.dtype.__get__`) instead of a single `get` check here. But I feel that might be cleaner on what we are trying to implement, I'll update.